modules/openxAnalyticsAdapter.js

```diff
@@ -364,94 +366,93 @@ function onSlotLoaded({ slot }) {
 
-openxAdapter.enableAnalytics = function(adapterConfig) {
-  analyticsConfig = {...DEFAULT_ANALYTICS_CONFIG, ...adapterConfig.options};
-  analyticsConfig.testCode = getTestCode();
-  analyticsConfig.utmTagData = this.buildUtmTagData();
-  utils.logInfo('OpenX Analytics enabled with config', analyticsConfig);
+  if(isValidConfig(adapterConfig)){
+    analyticsConfig = {...DEFAULT_ANALYTICS_CONFIG, ...adapterConfig.options};
+    analyticsConfig.utmTagData = this.buildUtmTagData();
+    utils.logInfo('OpenX Analytics enabled with config', analyticsConfig);
+
+    if (analyticsConfig.testPipeline) {
+      openxAdapter.track = (args) => {
+        prebidAnalyticsEventHandlerV1(args);
+        prebidAnalyticsEventHandlerV2(args);
+      };
+
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, args => {
+          utils.logInfo('OX: SlotOnLoad event triggered');
+          onSlotLoaded(args);
+          onSlotLoadedV2(args);
+        });
+      });
 
-  if (analyticsConfig.testPipeline) {
-    // override track method with v2 handlers
-    openxAdapter.track = prebidAnalyticsEventHandler;
+    } else if (analyticsConfig.enableV2) {
+      // override track method with v2 handlers
+      openxAdapter.track = prebidAnalyticsEventHandlerV2;
 
-    googletag.cmd.push(function() {
-      googletag.pubads().addEventListener(SLOT_LOADED, args => {
-        openxAdapter.track({ eventType: SLOT_LOADED, args });
-        utils.logInfo('OX: SlotOnLoad event triggered');
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, args => {
+          openxAdapter.track({ eventType: SLOT_LOADED, args });
+          utils.logInfo('OX: SlotOnLoad event triggered');
+        });
       });
-    });
-  } else {
-    googletag.cmd.push(function() {
-      googletag.pubads().addEventListener(SLOT_LOADED, function(args) {
-        utils.logInfo('OX: SlotOnLoad event triggered');
-        onSlotLoaded(args);
+    } else {
+      openxAdapter.track = prebidAnalyticsEventHandlerV1;
+      googletag.cmd.push(function() {
+        googletag.pubads().addEventListener(SLOT_LOADED, function(args) {
+          utils.logInfo('OX: SlotOnLoad event triggered');
+          onSlotLoaded(args);
+        });
       });
-    });
+    }
+
+    openxAdapter.originEnableAnalytics(adapterConfig);
   }
 
-  openxAdapter.originEnableAnalytics(adapterConfig);
+  function isValidConfig({options: analyticsOptions}){
+    const fieldValidations = [
+      // tuple of property, type, required
+      ['publisherPlatformId', 'string', true],
+      ['publisherAccountId', 'number', true],
+      ['sampling', 'number', false],
+      ['enableV2', 'boolean', false],
+      ['testPipeline', 'boolean', false],
+      ['adIdKey', 'string', false],
+      ['payloadWaitTime', 'number', false],
+    ];
+
+    let failedValidation = fieldValidations.find(([property, type, required]) => {
```

Comment:
Sometimes I wonder if it really makes sense to use `utils.find` or just the native find from ES6 since we do some backfilling for that through `@babel/env`

Reply:
This should be the native ES6 find.  The only reason why I use the util._each/_map is that it works on objects whereas the native one does not.